pkg/cli/debug/debug.go

```diff
@@ -658,6 +677,12 @@ func (o *DebugOptions) transformPodForDebug(annotations map[string]string) (*cor
 		container.SecurityContext.RunAsNonRoot = nil
 	}
 
+	// When run in another namespace than original pod, any initContainers that run a command
+	// that depends on VolumeMounts will fail, so unless the initContainer is explicitly being
+	// debugged, remove the initContainers.
+	if len(o.ToNamespace) > 0 && !isInitContainer(pod, o.Attach.ContainerName) {
```

Comment:
This is too broad a check.  Only init containers that depend on volumes that were stripped are impacted.  Many init containers use empty dir, none of those are affected.

So the logic should be when stripping volumes is to identify the containers that referenced those mounts, and only remove those.